packages/eslint-plugin/tests/rules/consistent-type-assertions.test.ts

```diff
@@ -85,6 +85,38 @@ print?.(<Foo>{ bar: 5 })
 print?.call(<Foo>{ bar: 5 })
 print\`\${<Foo>{ bar: 5 }}\`
 `;
+const ARRAY_LITERAL_AS_CASTS = `
+const x = [] as string[];
+const x = ['a'] as string[];
+const x = [] as Array<string>;
+const x = ['a'] as Array<string>;
+const x = [Math.random() ? 'a' : 'b'] as 'a'[];
+`;
+const ARRAY_LITERAL_ANGLE_BRACKET_CASTS = `
+const x = <string[]>[];
+const x = <string[]>['a'];
+const x = <Array<string>>[];
+const x = <Array<string>>['a'];
+const x = <'a'[]>[Math.random() ? 'a' : 'b'];
+`;
+const ARRAY_LITERAL_ARGUMENT_AS_CASTS = `
+print([5] as Foo);
+new print([5] as Foo);
+function foo() { throw [5] as Foo }
+function b(x = [5] as Foo.Bar) {}
+function c(x = [5] as Foo) {}
+print?.([5] as Foo);
+print?.call([5] as Foo);
+print\`\${[5] as Foo}\`;
+`;
+const ARRAY_LITERAL_ARGUMENT_BRACKET_CASTS = `
+print(<Foo>[5]);
+new print(<Foo>[5]);
+function foo() { throw <Foo>[5] }
+print?.(<Foo>[5]);
+print?.call(<Foo>[5]);
+print\`\${<Foo>[5]}\`;
+`;
```

Comment:
😬 I know this is existing code with its own pattern, but - we're trying to move away from auto-generated tests. They're much harder to statically analyze and tend to batch together a lot of redundant tests with multiple edge cases per test. Could you please keep to manually written tests for the new code?

Since all this change does functionally is add a new typed of checked assertion (arrays), I think adding a test per each of these lines should be fine.

Reply:
Sounds good, 

>  I think adding a test per each of these lines should be fine.

Does this mean I can write something like this?
```ts
    {
      code: `
const x = [] as string[];
const x = ['a'] as string[];
const x = [] as Array<string>;
const x = ['a'] as Array<string>;
const x = [Math.random() ? 'a' : 'b'] as 'a'[];
      `, options: [
        {
          assertionStyle: "as"
        }
      ]
    }
```